src/org/opendatakit/briefcase/model/BriefcaseAnalytics.java

```diff
@@ -0,0 +1,182 @@
+package org.opendatakit.briefcase.model;
+
+import com.brsanthu.googleanalytics.EventHit;
+import com.brsanthu.googleanalytics.GoogleAnalytics;
+import com.brsanthu.googleanalytics.GoogleAnalyticsConfig;
+import org.opendatakit.briefcase.buildconfig.BuildConfig;
+
+/**
+ *  This class is used to send usage information to ODK's Google Analytics account.
+ *
+ * <P>A library built by 'brsanthu' is utilized to streamline the
+ * process of contacting Google's Measurement Protocol API.</P>
+ * <P>This class also implements the Singleton pattern.</P>
+ *
+ * @author Joao Paulo Knox
+ * @see <a href="https://github.com/brsanthu/google-analytics-java">google-analytics-java</a>
+ */
+public class BriefcaseAnalytics {
+
+  private static final String GOOGLE_TRACKING_ID = BriefcasePreferences.GOOGLE_TRACKING_ID;
+  private static final String UNIQUE_USER_ID = BriefcasePreferences.getUniqueUserID();
+  private static final String EVENT_CATEGORY = "ODK Briefcase";
+
+  private static BriefcaseAnalytics instance = null;
+  private static GoogleAnalytics googleAnalytics = null;
+  private static GoogleAnalyticsConfig config = null;
+  private static Boolean briefcaseTrackingConsent = null;
+  private static Boolean hasTrackedStartup = false;
+
+  /**
+   * Constructor
+   *
+   * <P>This constructor will instantiate 'brsanthu's {@link GoogleAnalytics} class and
+   * check if the user is okay with their behaviour being tracked.</P>
+   * <P>The constructor is private so as to adhere to the rules of the Singleton pattern.</P>
+   */
+  private BriefcaseAnalytics() {
+    config = new GoogleAnalyticsConfig();
+    checkTrackingConsentState();
+    this.googleAnalytics = new GoogleAnalytics(config, GOOGLE_TRACKING_ID);
+  }
+
+  /**
+   * Singleton creator.
+   *
+   * <P>This method (along with the private constructor) will ensure
+   * that only one instance of this class is instantiated.</P>
+   * @return the instantiated Singleton of {@link BriefcaseAnalytics}.
+   */
+  public static BriefcaseAnalytics getInstance() {
+    return instance == null ? new BriefcaseAnalytics() : instance;
```

Comment:
That's pretty cool, Eclipse was good albeit my time with it has helped me to really appreciate IntelliJ.